java/maven/src/org/netbeans/modules/maven/api/output/OutputUtils.java

```diff
@@ -263,9 +263,12 @@ protected ClassPath getClassPath() {
             Project prj = ref.get();
             if(prj != null) {
                 ProjectSourcesClassPathProvider prov = prj.getLookup().lookup(ProjectSourcesClassPathProvider.class);
+                // appending compile cp probably doesn't hurt although everything should be in exec cp already
                 return createProxyClassPath(
-                    Stream.concat(Stream.of(prov.getProjectClassPaths(ClassPath.EXECUTE)),
-                                  Stream.of(prov.getProjectClassPaths(ClassPath.BOOT)))
+                    Stream.of(Stream.of(prov.getProjectClassPaths(ClassPath.EXECUTE)),
+                              Stream.of(prov.getProjectClassPaths(ClassPath.COMPILE)),
+                              Stream.of(prov.getProjectClassPaths(ClassPath.BOOT)))
+                          .flatMap(s -> s)
```

Comment:
everything should be in the exec cp I think, but https://github.com/apache/netbeans/blob/139498df4577f8cfb1b6c76a1c30a4ca8002ea8b/java/maven/src/org/netbeans/modules/maven/classpath/ClassPathProviderImpl.java#L287 is difficult to debug due to all the lazy lambdas.

If tests have a module info (and src too), test sources seem to be missing atm.